src/mesh/rectilineargrid.jl

```diff
@@ -35,12 +35,13 @@ vertex(g::RectilinearGrid{Dim}, ijk::Dims{Dim}) where {Dim} = Point(getindex.(g.
 
 xyz(g::RectilinearGrid) = g.xyz
 
-@generated function XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T}
+XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T} = xyzXYZ(T, xyz(g))
+
+@generated function xyzXYZ(::Type{T}, xyz::NTuple{Dim}) where {T,Dim}
```

Comment:
Maybe this code could be like this:
```suggestion
@generated function xyzXYZ(xyz::NTuple{Dim,<:AbstractVector{T}}) where {Dim,T}
```